optuna/pruners/percentile.py

```diff
@@ -192,3 +192,10 @@ def prune(self, study, trial):
         if direction == StudyDirection.MAXIMIZE:
             return best_intermediate_result < p
         return best_intermediate_result > p
+
+    def is_target_step(self, step):
+        # type: (int) -> bool
+        if step < self._n_warmup_steps:
+            return False
+
+        return (step - self._n_warmup_steps) % self._interval_steps == 0
```

Comment:
Please add a test case to confirm if the target steps determined by this function matches with those determined by the `prune` method? I think the test cases can be written using the [`DeteministicPruner(True)`](https://github.com/optuna/optuna/blob/master/optuna/testing/integration.py#L4-L13).

IMO, the current logic does not match with `prune` in some cases. For example, the check corresponding to L198 seems to be `if step <= self._nwarmup_steps` in the `prune` method.